lib/widgets/mxc_image.dart

```diff
@@ -182,26 +182,33 @@ class _MxcImageState extends State<MxcImage> {
     }
 
     if (event != null) {
-      if (!PlatformInfos.isWeb) {
-        final fileInfo = await event.getFileInfo(
+      try {
+        if (!PlatformInfos.isWeb) {
+          final fileInfo = await event.getFileInfo(
+            getThumbnail: widget.isThumbnail,
+          );
+          if (fileInfo != null && fileInfo.filePath.isNotEmpty) {
+            setState(() {
+              filePath = fileInfo.filePath;
+            });
+            return;
+          }
+        }
+
+        final matrixFile = await event.downloadAndDecryptAttachment(
           getThumbnail: widget.isThumbnail,
         );
-        if (fileInfo != null && fileInfo.filePath.isNotEmpty) {
-          setState(() {
-            filePath = fileInfo.filePath;
-          });
-          return;
-        }
+        if (!mounted) return;
+        _imageData = matrixFile.bytes;
+        return;
+      } catch (e) {
+        Logs().e('MxcImage::Error while downloading image: $e');
+      } finally {
+        setState(() {
+          isLoadDone = true;
+        });
```

Comment:
IMO: If the case has error, still go on it. This func only used for success case, pls check it again